### PR TITLE
OT-85 Update ApplicationDcContext to latest core

### PR DESCRIPTION
**Pull request by Boehrsi**
Describe what the problem was / what the new feature is
New core version with new handling, e.g for strings.

**Describe your solution**

- Updated Gradle
- Moved all handler classes to the **handler** package
- Removed ApplicationDcContext.java and created the NativeInteractionManager.java which handles the relevant cases for the OX Talk app (for now, additional handling for push / background / network will be added later, probably directly in dart)
- Removed the MessageChannel for error, now the EventChannel system is used
- Updated the Dc* classes

**Additional context**
Add any other context about the bug / feature here.

### CODE REVIEW

File: android/src/main/java/com/openxchange/deltachatcore/DeltaChatCorePlugin.java [hunk 1]
```diff
@@ -185,13 +193,13 @@ private void coreListener(MethodCall call, Result result) {
...
         if (eventHandlers.indexOfKey(eventId) < 0) {
-            eventHandler = new EventChannelHandler(applicationDcContext, registrar.messenger(), eventId);
+            eventHandler = new EventChannelHandler(nativeInteractionManager, registrar.messenger(), eventId);
             eventHandlers.put(eventId, eventHandler);
         } else {
             eventHandler = eventHandlers.get(eventId);
         }
         if (add) {
-            int newObserverId = eventHandler.addObserver();
+            int newObserverId = eventHandler.addListener();
```
Comment (florianhaar) [inline android/src/main/java/com/openxchange/deltachatcore/DeltaChatCorePlugin.java]: Maybe 'newListenerId' is a better wording than 'newObserverId'.

**Review by florianhaar — APPROVED** [review]
Just one idea. The rest looks good and could be merged.

File: android/src/main/java/com/openxchange/deltachatcore/DeltaChatCorePlugin.java (hunk 1 above)
Comment (Boehrsi) [reply android/src/main/java/com/openxchange/deltachatcore/DeltaChatCorePlugin.java]: Fixed by https://github.com/open-xchange/flutter-deltachat-core/pull/8/commits/5cc17937d9a06c1632d1e6bea958dad2712a3667

**Comment by Boehrsi** [comment]
Added review fixes and merged the ticket.

**Pull request merged**